docs/filtering/dehardsubbing.md

```diff
@@ -0,0 +1,222 @@
+to replace whole frames like this if it can be avoided. A better solution then
+would be to replace only the hardsubbed region of the scene. This can be
+accomplished with classes like `HardsubSign`:
+
+```py
+from vsmasktools import HardsubSign
+
+dehardsub = HardsubSign(ranges=[(1138, 1221)]).apply_dehardsub(hardsub, ref)
+```
+
+This will find the differences between the `hardsub` and `ref` clips, for the
+same frame range as before, and apply a mask that replaces just the hardsubbed
+region of `hardsub` with the respective region from `ref`.
+
+=== "Hardsubbed"
+    ![](../static/dehardsubbing-simple-hardsub.png)
+
+=== "Hardsub mask"
+    ![](../static/dehardsubbing-simple-mask.png)
+
+## Bounding
+
+This next example demonstrates two kinds of hardsubs: the good and the bad.
+Here, the news broadcast's overlay is actually good (for an English release, at
+least) and leaving it hardsubbed would save a typesetter some time. However,
+this scene also has the bad Arial signs that could be redone better in softsubs
+if they're important. If one tries to use `HardsubSign` like before on this, it
+will mask everything.
+
+=== "Hardsubbed"
+    ![](../static/dehardsubbing-bound-hardsub.png)
+
+=== "Reference"
+    ![](../static/dehardsubbing-bound-ref.png)
+
+=== "Hardsub mask"
+    ![](../static/dehardsubbing-bound-mask.png)
+
+Since the bad hardsubs are all contained to a single region for the whole scene,
+the mask can be bounded by a `BoundingBox`. Think of this as masking the mask.
+
+```py
+from vsmasktools import BoundingBox, HardsubSign
+
+dehardsub = HardsubSign(
+    ranges=[(13784, 13843)],
+    bound=BoundingBox(pos=(0, 0), size=(1230, 880))
+).apply_dehardsub(hardsub, ref)
+```
+
+The bound is drawn from coordinate (0, 0) (the top-left corner) to a size of
+1230x880, which is enough to cover only the Arial signs.
+
+=== "Hardsubbed"
+    ![](../static/dehardsubbing-bound-hardsub.png)
+
+=== "Hardsub mask"
+    ![](../static/dehardsubbing-bound-mask.png)
+
+=== "Bounded hardsub mask"
+    ![](../static/dehardsubbing-bound-bounded-mask.png)
+
+=== "Dehardsubbed"
+    ![](../static/dehardsubbing-bound-dehardsub.png)
+
+## Fades
+
+Hardsubs that fade in or out may require special handling to account for when
+the hardsub is at its peak transparency. Lowering the threshold parameter `thr`
+of `HardsubSign` may help but this can also begin to catch differences in noise
+between the clips. This example will use `HardsubSignFades`, a specialized
+version of `HardsubSign`, on the first frame of a fade-in since that is the most
+difficult part of the sequence to mask.
+
+```py
+from vsmasktools import HardsubSignFades
+from vstools import get_y, replace_ranges
+
+dehardsub_mask = HardsubSignFades().get_mask(hardsub, ref)
+dehardsub = hardsub.std.MaskedMerge(ref, get_y(dehardsub_mask))
+dehardsub = replace_ranges(hardsub, dehardsub, (1462, 1524))
+```
+
+=== "Hardsubbed"
+    ![](../static/dehardsubbing-fade.png)
```

Comment:
Considered doing a low-fps gif here to visualize the fade but felt that a single frame would be easier to follow 🤷